loader/loader.c

```diff
@@ -6049,6 +6028,19 @@ VkResult setup_loader_term_phys_devs(struct loader_instance *inst) {
                 if (res == VK_ERROR_OUT_OF_HOST_MEMORY) {
                     goto out;
                 }
+
+                loader_set_dispatch((void *)new_phys_devs[idx], inst->disp);
+                new_phys_devs[idx]->this_icd_term = icd_phys_dev_array[i].icd_term;
+                new_phys_devs[idx]->icd_index = (uint8_t)(icd_phys_dev_array[i].icd_index);
+                new_phys_devs[idx]->phys_dev = icd_phys_dev_array[i].physical_devices[j];
+
+                // Fill in the properties
+                new_phys_devs[idx]->this_icd_term->dispatch.GetPhysicalDeviceProperties(new_phys_devs[idx]->phys_dev,
+                                                                                        &new_phys_devs[idx]->properties);
+
+                // Fill in driver support of loader-interested device extensions
+                device_extensions_supported_by_physical_device(new_phys_devs[idx]);
```

Comment:
Again, this probably should be moved into its own loop. That way it is applied to all physical devices and none are forgotten.